test_conformance/SVM/test_allocate_shared_buffer_negative.cpp

```diff
@@ -0,0 +1,102 @@
+//
+// Copyright (c) 2017 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "common.h"
+
+const cl_mem_flags svm_flag_set[] = {
+    CL_MEM_READ_WRITE,
+    CL_MEM_WRITE_ONLY,
+    CL_MEM_READ_ONLY,
+    CL_MEM_READ_WRITE | CL_MEM_SVM_FINE_GRAIN_BUFFER,
+    CL_MEM_WRITE_ONLY | CL_MEM_SVM_FINE_GRAIN_BUFFER,
+    CL_MEM_READ_ONLY | CL_MEM_SVM_FINE_GRAIN_BUFFER,
+    CL_MEM_READ_WRITE | CL_MEM_SVM_FINE_GRAIN_BUFFER | CL_MEM_SVM_ATOMICS,
+    CL_MEM_WRITE_ONLY | CL_MEM_SVM_FINE_GRAIN_BUFFER | CL_MEM_SVM_ATOMICS,
+    CL_MEM_READ_ONLY | CL_MEM_SVM_FINE_GRAIN_BUFFER | CL_MEM_SVM_ATOMICS,
+    0
+};
+const char* svm_flag_set_names[] = {
+    "CL_MEM_READ_WRITE",
+    "CL_MEM_WRITE_ONLY",
+    "CL_MEM_READ_ONLY",
+    "CL_MEM_READ_WRITE | CL_MEM_SVM_FINE_GRAIN_BUFFER",
+    "CL_MEM_WRITE_ONLY | CL_MEM_SVM_FINE_GRAIN_BUFFER",
+    "CL_MEM_READ_ONLY | CL_MEM_SVM_FINE_GRAIN_BUFFER",
+    "CL_MEM_READ_WRITE | CL_MEM_SVM_FINE_GRAIN_BUFFER | CL_MEM_SVM_ATOMICS",
+    "CL_MEM_WRITE_ONLY | CL_MEM_SVM_FINE_GRAIN_BUFFER | CL_MEM_SVM_ATOMICS",
+    "CL_MEM_READ_ONLY | CL_MEM_SVM_FINE_GRAIN_BUFFER | CL_MEM_SVM_ATOMICS",
+    "0"
+};
+
+
+int test_svm_allocate_shared_buffer_negative(cl_device_id deviceID,
+                                             cl_context context2,
+                                             cl_command_queue queue,
+                                             int num_elements)
+{
+    clContextWrapper context = NULL;
```

Comment:
Not a must-fix since this follows the same pattern used by other SVM tests, but the multiple contexts (and multiple queues) confused me initially.  I would consider renaming the function arguments to something like `context_from_harnes` or `queue_from_harness` so it's more obvious that they should not be used, or perhaps to use the single-device context (and queue?) as-is for the tests where we do not need a multi-device context (are there any of these?).